contracts/utils/OrderPayable.sol

```diff
@@ -0,0 +1,167 @@
+                IAccessControlRegistry(accessControlRegistry).hasRole(
+                    orderSignerRole,
+                    orderSigner
+                ),
+            "Invalid order signer"
+        );
+        require(paymentAmount > 0, "Payment amount zero");
+        require(!orderIdToPaymentStatus[orderId], "Order already paid for");
+        _;
+    }
+
+    constructor(
+        address _accessControlRegistry,
+        string memory _adminRoleDescription,
+        address _manager
+    )
+        AccessControlRegistryAdminnedWithManager(
+            _accessControlRegistry,
+            _adminRoleDescription,
+            _manager
+        )
+    {
+        orderSignerRole = _deriveRole(
+            _deriveAdminRole(manager),
+            ORDER_SIGNER_ROLE_DESCRIPTION
+        );
+        withdrawerRole = _deriveRole(
+            _deriveAdminRole(manager),
+            WITHDRAWER_ROLE_DESCRIPTION
+        );
+    }
+
+    function payForOrder(
+        bytes32 orderId,
+        uint256 expirationTimestamp,
+        address orderSigner,
+        bytes calldata signature
+    )
+        external
+        payable
+        onlyValidOrderParameters(
+            orderId,
+            expirationTimestamp,
+            orderSigner,
+            msg.value
+        )
+    {
+        require(
+            (
+                keccak256(
+                    abi.encodePacked(
+                        block.chainid,
+                        address(this),
+                        orderId,
+                        expirationTimestamp,
+                        msg.value
+                    )
+                ).toEthSignedMessageHash()
+            ).recover(signature) == orderSigner,
+            "Signature mismatch"
+        );
+        orderIdToPaymentStatus[orderId] = true;
+        emit PaidForOrder(orderId, expirationTimestamp, orderSigner, msg.value);
+    }
+
+    function payInvoice(
+        bytes32 orderId,
+        uint256 expirationTimestamp,
+        address orderSigner,
+        bytes calldata signature
+    )
+        external
+        payable
+        onlyValidOrderParameters(
+            orderId,
+            expirationTimestamp,
+            orderSigner,
+            msg.value
+        )
+    {
+        require(
+            (
+                keccak256(
+                    abi.encodePacked(
+                        block.chainid,
+                        address(this),
+                        orderId,
+                        expirationTimestamp,
+                        msg.sender,
+                        msg.value
+                    )
+                ).toEthSignedMessageHash()
+            ).recover(signature) == orderSigner,
+            "Signature mismatch"
+        );
+        orderIdToPaymentStatus[orderId] = true;
+        emit PaidInvoice(
+            orderId,
+            expirationTimestamp,
+            orderSigner,
+            msg.sender,
+            msg.value
+        );
+    }
```

Comment:
@RiVal-cz wants to axe this for the initial version

Reply:
Removed [here](https://github.com/api3dao/airnode-protocol-v1/pull/128/commits/ea1a4f3c59f68acc5155d7c5d7c721cf48bf9d6c)